File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        int s=1;
        ArrayList<Integer> l = new ArrayList<>();
        for(int n:nums){
            if(n%2==1){
                l.add(s);
                s=1;
            }else s++;
        }
        l.add(s);
        int ans=0;
        for(int i=k;i<l.size();i++){
            ans+=l.get(i)*l.get(i-k);
        }
        return ans;
    }
}

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        s=1
        l=[]
        for n in nums:
            if n&1:
                l.append(s)
                s=0
            s+=1
        l.append(s)
        ans = 0
        for x,y in zip(l,l[k:]):
            ans+=x*y
        return ans